apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1048,7 +1048,9 @@ call_query(QM, Id, Index, Ref, Query, QueryOpts) ->
             ?RESOURCE_ERROR(not_found, "resource not found")
     end.
 
-do_call_query(QM, Id, Index, Ref, Query, #{is_buffer_supported := true} = QueryOpts, Resource) ->
+do_call_query(QM, Id, Index, Ref, Query, QueryOpts, #{query_mode := ResQM} = Resource) when
```

Comment:
do the authn authz resources end up calling this path?

Reply:
Yes, they end up calling this function.

Reply:
Here I introduce a new query mode type: #11724 
Perhaps it would be more appropriate to restrict this behavior to that more specific type.

Reply:
But I think we would also need to introduce a new query mode for the async version of a buffered connector, to have the correct semantics as well.  :thinking:

Reply:
https://github.com/emqx/emqx/pull/11724/commits/5467e32b9c9b1126b819d79c61787b1d3e2dccf6